### PR TITLE
UpdateAccount & Delete Account /claims

### LINKED ISSUE #61: Secure backend
Secure backend so nobody can change data without signed in etc

### PR DESCRIPTION
**Pull request by krapluka**
Implemented Endpoints:
- Update User
- Delete User
- fixed some claims issues and secured endpoints

### CODE REVIEW

File: Backend/ActiveCruzer/Controllers/UserController.cs [hunk 1]
```diff
@@ -125,6 +130,53 @@ public ActionResult<JwtDto> Login([FromBody] CredentialsDTO credentials)
...
             return Unauthorized();
         }
+        /// <summary>
+        /// Delete user account
+        /// </summary>
+        /// <param name="userId"></param>
+        /// <returns></returns>
+        [Authorize]
+        [HttpDelete]
+        [Route("Delete")]
```
Comment (andreasziegl) [inline Backend/ActiveCruzer/Controllers/UserController.cs:142]: please specify the possible response codes

File: Backend/ActiveCruzer/Controllers/UserController.cs [hunk 2]
```diff
@@ -125,6 +130,53 @@ public ActionResult<JwtDto> Login([FromBody] CredentialsDTO credentials)
...
+            }
+            return Unauthorized("You are not allowed to perform this action.");
+        }
+
+        /// <summary>
+        /// Update user
+        /// </summary>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        [Authorize]
```
Comment (andreasziegl) [inline Backend/ActiveCruzer/Controllers/UserController.cs:161]: please specify the possible response codes

File: Backend/ActiveCruzer/Controllers/UserController.cs [hunk 3]
```diff
@@ -125,6 +130,53 @@ public ActionResult<JwtDto> Login([FromBody] CredentialsDTO credentials)
...
+                {
+                    return Ok(_user);
+                }
+                return Unauthorized("You are not allowed to perform this action.");
+        }
+
+        /// <summary>
+        /// get user account from logged in user
+        /// </summary>
+        /// <returns></returns>
```
Comment (andreasziegl) [inline Backend/ActiveCruzer/Controllers/UserController.cs:177]: please specify the possible response codes

**Review by andreasziegl — CHANGES_REQUESTED** [review]

File: Backend/ActiveCruzer/Controllers/UserController.cs (hunk 1 above)
Comment (krapluka) [reply Backend/ActiveCruzer/Controllers/UserController.cs:142]: done

**Review by andreasziegl — APPROVED** [review]

**Pull request merged**